packages/app/components/Settings/CourseRoster.tsx

```diff
@@ -0,0 +1,122 @@
+import { SearchOutlined } from "@ant-design/icons";
+import { API } from "@koh/api-client";
+import { Role } from "@koh/common";
+import { Divider, Input, List, Pagination, Spin } from "antd";
+import Avatar from "antd/lib/avatar/avatar";
+import { useState } from "react";
+import { ReactElement } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+type CourseRosterProps = { courseId: number };
+
+type RenderTableProps = {
+  courseId: number;
+  role: Role;
+  listTitle: string;
+  displaySearchBar: boolean;
+  searchPlaceholder: string;
+};
+
+const CourseRosterComponent = styled.div`
+  margin-left: auto;
+  margin-right: auto;
+`;
+
+export default function CourseRoster({
+  courseId,
+}: CourseRosterProps): ReactElement {
+  return (
+    <CourseRosterComponent>
+      <h1>Course Roster</h1>
+      <RenderTable
+        courseId={courseId}
+        role={Role.PROFESSOR}
+        listTitle={"Professors"}
+        displaySearchBar={false}
+        searchPlaceholder=""
+      />
+      <br />
+      <RenderTable
+        courseId={courseId}
+        role={Role.TA}
+        listTitle={"Teaching Assistants"}
+        displaySearchBar={true}
+        searchPlaceholder="Search TAs"
+      />
+      <br />
+      <RenderTable
+        courseId={courseId}
+        role={Role.STUDENT}
+        listTitle={"Students"}
+        displaySearchBar={true}
+        searchPlaceholder="Search students"
+      />
+      <br />
+      <Divider />
+    </CourseRosterComponent>
+  );
+}
+
+function RenderTable({
+  courseId,
+  role,
+  listTitle,
+  displaySearchBar,
+  searchPlaceholder,
+}: RenderTableProps): ReactElement {
+  const [page, setPage] = useState(1);
+  const [search, setSearch] = useState("");
+  const handleSearch = (event) => {
+    event.preventDefault();
+    setSearch(event.target.value);
+    setPage(1);
+  };
+  const { data } = useSWR(
+    `${role}/${page}/${search}`,
+    async () => await API.course.getUserInfo(courseId, page, role, search)
+  );
+  if (!data) {
+    return <Spin tip="Loading..." size="large" />;
+  } else {
+    return (
+      <div style={{ backgroundColor: "white" }}>
+        <div style={{ backgroundColor: "#f0f0f0", height: "56px" }}>
+          <h3 style={{ position: "relative", left: "10px", top: "14px" }}>
+            {listTitle}
+          </h3>
+        </div>
+        {displaySearchBar && (
+          <Input
+            placeholder={searchPlaceholder}
+            prefix={<SearchOutlined />}
+            value={search}
+            onChange={handleSearch}
+          />
+        )}
+        <List
+          dataSource={data.users}
+          renderItem={(item) => (
+            <List.Item key={item.id}>
+              <List.Item.Meta
+                avatar={<Avatar src={item.photoURL} />}
```

Comment:
it would be nice to reuse the AvatarWithInitals component (so if they dont have a photo we render their initials) 
<img width="111" alt="image" src="https://user-images.githubusercontent.com/59672089/172969304-318a545c-1d6e-4e2e-addd-fde5038f44e6.png">

Reply:
Just added that option